federation/src/main/scala/caliban/federation/FederationHelpers.scala

```diff
@@ -6,23 +6,24 @@ import caliban.Value.{ NullValue, StringValue }
 import caliban.introspection.adt.__InputValue
 import caliban.schema.{ ArgBuilder, Schema, Types }
 
+import scala.collection.mutable.ListBuffer
+
 private[federation] object FederationHelpers {
-  def traverseEither[A, B](list: List[Either[A, B]]): Either[A, List[B]] = {
-    val iterator = list.iterator
-    val result   = List.newBuilder[B]
-    var error    = Option.empty[A]
+  import caliban.syntax._
+
+  private def traverseEither[A, B](list: List[Either[A, B]]): Either[A, List[B]] = {
```

Comment:
I'm wondering if this should actually be an iterable. Then we could use a lazily evaluated view over the incoming collection to improve perf for cases where it fails early in the traverse

Reply:
I kept it as a list but changed it to `List[InputValue]` and moved `anyArgBuilder.build` within the method instead. Since this method is used only in a single place where we already have the `List[InputValue]` materialized I think this makes the most sense performance-wise.